src/org/zaproxy/zap/extension/ascanrulesAlpha/TraceAxdScanner.java

```diff
@@ -0,0 +1,184 @@
+
+import org.apache.commons.httpclient.URI;
+import org.apache.commons.httpclient.URIException;
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.core.scanner.AbstractAppPlugin;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.core.scanner.Category;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpRequestHeader;
+import org.parosproxy.paros.network.HttpStatusCode;
+import org.zaproxy.zap.model.Tech;
+import org.zaproxy.zap.model.TechSet;
+
+import org.parosproxy.paros.Constant;
+
+import java.io.IOException;
+
+/**
+ * Active scan rule which checks whether or not trace.axd is exposed.
+ * https://github.com/zaproxy/zaproxy/issues/3280
+ * 
+ * @author kingthorin+owaspzap@gmail.com
+ */
+public class TraceAxdScanner extends AbstractAppPlugin {
+
+	private static final String MESSAGE_PREFIX = "ascanalpha.traceaxdscanner.";
+	private static final int PLUGIN_ID = 40029;
+
+	private static final Logger LOG = Logger.getLogger(TraceAxdScanner.class);
+
+	@Override
+	public int getId() {
+		return PLUGIN_ID;
+	}
+
+	@Override
+	public String getName() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "name");
+	}
+
+	@Override
+	public String getDescription() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "desc");
+	}
+
+	@Override
+	public String getSolution() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "soln");
+	}
+
+	@Override
+	public String getReference() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "refs");
+	}
+	
+	private String getOtherInfo() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "otherinfo");
+	}
+
+	@Override
+	public String[] getDependency() {
+		return null;
+	}
+
+	@Override
+	public boolean targets(TechSet technologies) {
+		return technologies.includes(Tech.IIS) || technologies.includes(Tech.Windows) || technologies.includes(Tech.ASP)
+				|| technologies.includes(Tech.MsSQL);
+	}
+
+	@Override
+	public int getCategory() {
+		return Category.INFO_GATHER;
+	}
+
+	@Override
+	public int getRisk() {
+		return Alert.RISK_MEDIUM;
+	}
+
+	@Override
+	public int getCweId() {
+		return 215; // CWE-215: Information Exposure Through Debug Information
+	}
+
+	@Override
+	public int getWascId() {
+		return 13; // WASC-13: Informatin Leakage
+	}
+
+	@Override
+	public void init() {
+
+	}
+
+	@Override
+	public void scan() {
+
+		// Check if the user stopped things. One request per URL so check before
+		// sending the request
+		if (isStop()) {
+			if (LOG.isDebugEnabled()) {
+				LOG.debug("Scanner " + getName() + " Stopping.");
+			}
+			return;
+		}
+
+		HttpMessage newRequest = getNewMsg();
+		newRequest.getRequestHeader().setMethod(HttpRequestHeader.GET);
+		URI baseUri = getBaseMsg().getRequestHeader().getURI();
+		URI elmahUri = null;
+		try {
+			String baseUriPath = baseUri.getPath();
+			String newPath = baseUriPath.endsWith("/") ? baseUriPath + "trace.axd" : baseUriPath + "/trace.axd"; 

```

Comment:
- This (`baseUriPath.endsWith`) will lead to a `NullPointerException` when the URL does not have a path (e.g. `http://example.com`).
 - Does it worth checking if the path contains a "file"? For example, for `http://example.com/Login.aspx` it will try `http://example.com/Login.aspx/trace.axd`.